tests/metrics/utils.py

```diff
@@ -104,14 +108,14 @@ def _class_test(
             if rank == 0:
```

Comment:
```suggestion
            if rank == 0 and check_dist_sync_on_step:
                ddp_preds = ...
                ddp_target = ...
                sk_batch_result = ..
                _assert_...
```
can be done like this since there is no else condition here?